Review

i-bocharov opened the pull request.

i-bocharov committed:
chore: update dependencies

- Ran 'npm install' to update project dependencies.

i-bocharov committed:
chore: add open: true to Webpack configuration

- Added 'open: true' to Webpack dev server settings.
- Ensures the browser automatically opens when the development server starts.

i-bocharov committed:
feat: add routing and reorganize project structure

- Added routing in the App component using <Routes> and <Route>:
  - Main routes: '/', '/feed', '/login', '/register', '/forgot-password', '/reset-password', '/profile', '/profile/orders'.
  - Protected routes for '/login', '/register', '/forgot-password', '/reset-password', '/profile', '/profile/orders' implemented using the ProtectedRoute component.
  - Modal routes: '/feed/:number', '/ingredients/:id', '/profile/orders/:number'.
  - Added '*' route to display the NotFound404 page.

- Reorganized project structure:
  - Renamed the 'services' folder to 'store'.
  - Moved files from the 'utils' folder (burger-api.ts, cookie.ts, types.ts) into a new shared folder:
    - burger-api.ts -> shared/api
    - types.ts -> shared/types
    - cookie.ts -> shared/utils
  - Updated aliases and imports according to the new structure.

i-bocharov committed:
feat: initialize redux store structure

- Add basic slices setup (ingredients, constructor, user)
- Create root reducer with initial states
- Setup store configuration

i-bocharov committed:
feat(ingredients): fetch and display ingredients from server

- Implement async slice for loading ingredients from the server
- Add selectors for buns, mains, sauces, and error from the store
- Update BurgerIngredients component to get data from Redux and display it
- Add error handling for ingredient loading
- Configure aliases and imports for hooks and selectors

i-bocharov committed:
feat(auth): protected routes setup

- Update ProtectedRoute component for route protection
- Implement auth check selectors
- Update base routing with protected routes
- Add redirect logic for authorized/unauthorized users

i-bocharov committed:
refactor(store): separate logic into slices and thunks

- Move async thunks to separate files
- Added order-slice
- Update user-slice with all auth operations support
- Split ingredients-product functionality into thunks and slice

i-bocharov committed:
fix: correct display of ingredients and constructor

- Updated app.tsx, ingredient-product-thunks.ts, ingredients-product-slice.ts, burger-ingredient.tsx — ingredients are now displayed correctly on the page
- Updated burger-constructor.tsx — removed undefined labels

i-bocharov committed:
feat: display ingredient details in a modal window

- Updated ingredient-details.tsx: added functionality to show ingredient details in a modal window when an ingredient is clicked.

i-bocharov committed:
feat: add ingredient to burger constructor

- Updated files constructor-product-slice.ts and burger-ingredients.tsx.
- Added functionality to include an ingredient in the constructor when the 'Add' button is clicked.

i-bocharov committed:
feat: implement order placement in burger constructor

- Added handling for the 'Place Order' button click and order submission.
- The button is disabled if no ingredients or bun are selected.

i-bocharov committed:
fix: properly close the order modal window

- The order modal now closes when clicking the close button, the overlay, or pressing Esc
- Added an action to reset order data when the modal is closed

i-bocharov committed:
fix: correct display of the orders feed

- Fixed Feed component: now properly returns FeedUI, added loading, error, and empty state checks
- Orders feed now displays correctly after receiving data from the store
- Moved thunk and slice for orders feed into separate files

i-bocharov committed:
fix: integrate store into Register, Login, OrderCard, and IngredientsCategory components

- Register and Login components now dispatch user data via corresponding thunks
- OrderCard and IngredientsCategory components use data from Redux store via useAppSelector

i-bocharov committed:
fix: correct loading and display of order data in the modal window

- Added getOrderByNumber call when opening the order modal
- Enhanced orders-slice to save orderModalData on successful order load
- Order modal now displays data after loading instead of only showing the preloader

i-bocharov committed:
feat: integrate store into Constructor, Profile, and Order History pages

- Connected Redux store to access loading state of ingredients, user, and user orders
- ConstructorPage, Profile, and ProfileOrders components now use useAppSelector to get data from the store

i-bocharov committed:
feat: implement and connect selectors for feed, orders, constructorProduct, and ingredients

- Added selectors for orders feed, user orders, constructor, and ingredients
- Components switched to using selectors to access data from Redux store
- Improved code readability and reusability

i-bocharov committed:
fix: proper handling of accessToken without double Bearer, fixed token storage, authorization headers, and fetching user orders

i-bocharov committed:
fix: add redirect to login page after logout

mikebars1995 commented on src/components/app-header/app-header.tsx:
```diff
@@ -1,4 +1,10 @@
 import { FC } from 'react';
 import { AppHeaderUI } from '@ui';
+import { useSelector } from 'react-redux';
```
**Нужно исправить**

* Хуки `useSelector` и `useDispatch` нужно импортировать из файла `store.ts` (они там типизированные уже), а не из библиотеки `react-redux`.  Нужно найти через поиск `react-redux` и заменить **все импорты**.

mikebars1995 reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Посмотрите на **гитхабе** все комментарии к коду (нужно прокрутить вниз страницу там)

Работа проделана огромная:




* Отлично, что в `prepare` создаете `id`
* Хорошая логика попапов
* Отлично сделана логика отображения попапа и страницы  под ним
* Хорошо сделали `ProtectedRoute`


но есть некоторые недочеты:


* По заданию нужно сделать логику **сортировки** ингредиентов внутри заказа (вверх и вниз).
* Хуки `useSelector` и `useDispatch` нужно импортировать из файла `store.ts` (они там типизированные уже), а не из библиотеки `react-redux`.  Нужно найти через поиск `react-redux` и заменить **все импорты**.
* Неавторизованный пользователь **не должен** отправлять заказ на сервер. При клике на кнопку его нужно переводить на логин и после удачного логина обратно к заказу.




Удачного рефакторинга кода.

i-bocharov committed:
refactor: replace useSelector with useAppSelector

- Updated all useSelector calls to useAppSelector for improved typing and better Redux integration.
- Changes do not affect logic, only enhance type safety and code readability.

i-bocharov committed:
feat: implement delete, sort, and reset of ingredients in burger constructor slice

- Add functionality to delete ingredients from the burger constructor state.
- Implement sorting of ingredients within the constructor.
- Add ability to reset ingredients to initial state.

i-bocharov committed:
feat: implement ingredient sorting, deletion, and constructor reset after order

- Added sorting functionality for ingredients within an order.
- Enabled deletion of ingredients from the burger constructor.
- Implemented automatic reset of the constructor state after order completion.

mikebars1995 reviewed (commented):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
